wrappers/rust/icicle-core/src/polynomials/mod.rs

```diff
@@ -0,0 +1,707 @@
+#[macro_export]
+macro_rules! impl_univariate_polynomial_api {
+    (
+        $field_prefix:literal,
+        $field_prefix_ident:ident,
+        $field:ident
+    ) => {
```

Comment:
There are no free-floating functions and everything is in the macro. This will mean that all the functions should be called from the curve/field crate directly instead of calling a core function with a field as a type parameter like we do for the other core primitives. What's the reason for doing it this way?
I guess the other way would cause a lot of free-floating functions

Reply:
I think the main difference is that this api is more object oriented, unlike functions such as MSM,NTT,vec_ops etc.
It's true that I don't have a generic Polynomial<F> but instead a polynomial trait that each field implements as a different type. I think it's mainly since the object is a shallow wrapper arond the C++ object so it doesn't really have what to do with this type.
In addition, other than init_cuda and init polynomials (From coeffs or evals) you use the polynomials directly (e.g. &a * &b) so you don't need to call from the curve really.

Please look at the exmaple examples/rust/polynomials/src/main.rs

Do you find that a problem?

Reply:
No, the more I thought about it the more it made sense. Just caught me off guard at first